elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/TableContext.java

```diff
@@ -0,0 +1,268 @@
+
+import com.yahoo.elide.datastores.aggregation.metadata.models.Argument;
+import com.yahoo.elide.datastores.aggregation.metadata.models.Table;
+import com.yahoo.elide.datastores.aggregation.query.Queryable;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLTable;
+
+import com.github.jknack.handlebars.EscapingStrategy;
+import com.github.jknack.handlebars.Handlebars;
+import com.github.jknack.handlebars.HandlebarsException;
+import com.github.jknack.handlebars.Helper;
+import com.github.jknack.handlebars.Options;
+import com.github.jknack.handlebars.Template;
+import org.apache.commons.lang3.StringUtils;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+/**
+ * TableContext for Handlebars Resolution.
+ */
+@Getter
+@ToString
+@Builder
+public class TableContext extends HashMap<String, Object> {
+
+    public static final String COL_PREFIX = "$$column";
+    public static final String REQ_PREFIX = "$$request";
+    public static final String TBL_PREFIX = "$$table";
+    public static final String ARGS_KEY = "args";
+    public static final String TABLE_KEY = "table";
+    public static final String COLUMNS_KEY = "columns";
+    public static final String NAME_KEY = "name";
+
+    private final MetaDataStore metaDataStore;
+
+    private final Queryable queryable;
+    private final String alias;
+
+    @Builder.Default
+    private final Map<String, Queryable> joins = new HashMap<>();
+
+    private final Handlebars handlebars = new Handlebars()
+                    .with(EscapingStrategy.NOOP)
+                    .registerHelper("sql", new Helper<Object>() {
+                        @Override
+                        public Object apply(final Object context, final Options options) throws IOException {
+                            return resolveSQLHandlebar(context, options);
+                        }
+                    });
+
+    public Object get(Object key) {
+
+        if (joins.containsKey(key)) {
+            Queryable joinQueryable = joins.get(key);
+            TableContext newCtx = TableContext.builder()
+                            .queryable(joinQueryable)
+                            .alias(appendAlias(alias, key.toString()))
+                            .metaDataStore(metaDataStore)
+                            .build();
+
+            joinQueryable.getColumnProjections().forEach(column -> {
+                newCtx.put(column.getName(), column.getExpression());
+            });
+
+            joinQueryable.getJoins().forEach((name, join) -> {
+                SQLTable joinTable = metaDataStore.getTable(join.getJoinTableType());
+                newCtx.addJoin(name, joinTable);
```

Comment:
Instead of storing the joins in the context, we can fetch them directly from the joinTable.